### DISCUSSION TITLE
How to format files automatically when `ctrl + s` saving

### DISCUSSION THREAD

**Discussion by a145789** [rome/tools#3757] [Q&A]
this is my vscode setting.json
```json
{
  "prettier.enable": false,
  "eslint.enable": false,
  "editor.formatOnSave": true,
  "editor.codeActionsOnSave": {
    "source.fixAll.rome": true
  },
  "[typescript]": {
    "editor.defaultFormatter": "rome.rome"
  },
  "files.eol": "\n"
}
```
but it didn't work, I have to use manual formatting,what can I do to make it automatically formatted.

**Comment by MichaReiser** [comment]
What's the name of the file that you're trying to format?

**Reply by a145789** [reply]
a simple `index.ts` file
For example

```ts
const foo = '';
// some useless \n







```

when I save, `\n` should normally be cleared, but I need to format it manually to clear these `\n`

**Reply by MichaReiser** [reply]
You won't see any change because Rome inserts a line break at the end of every file. 

Can you try with

```javascript
const      fooo = 
	'';
```

**Reply by a145789** [reply]
oh, I see, `rome` helped me format the file, but when I have many line break, `rome` doesn't format down to one

**Reply by MichaReiser** [reply]
Rome keeps at most one empty line between statements as you can see in this [example](https://docs.rome.tools/playground/?indentStyle=space&quoteStyle=single&trailingComma=none&code=YwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAIgB0AGUAcwB0ACIAKQA7AAoACgAKAAoAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAIgBsAG8AZwAiACkAOwA%3D)

Can you share an example with many empty lines that rome does not remove?

**Reply by a145789** [reply]
you can see this [repo](https://github.com/a145789/rome-format-line-break)

**Reply by MichaReiser** [reply]
I cloned your repo and hit save in the `index.ts` file. Rome formats the file as 

```
export const foo = 0;

```

removing all line breaks at the end except one

**Reply by a145789** [reply]
![a](https://user-images.githubusercontent.com/37403253/202206447-de98236d-a058-43a6-aa17-02716b161b2c.gif)

Well, I tried all night, and the final effect is still like this

Anyway, thank you for your reply

**Reply by MichaReiser** [reply]
That's the expected behavior. Rome removes all line breaks but leaves one at the end of the file.

**Reply by a145789** [reply]
take a closer look at this gif. It consists of two actions. First, `Rome` formats the file correctly, which is what I want, leaving only one line break.

Then I typed `git init` into the `Terminal` and `Rome` didn't work.

maybe I should create a bug issues

**Reply by a145789** [reply]
ok, I found out where the problem is, because my `vscode` configuration, in my `vscode settings.json`
```json
{
  // ...
  "editor.formatOnSaveMode": "modificationsIfAvailable"
  // ...
}
``` 

Everything works fine when `editor.formatOnSaveMode` is set to `file`

Thanks again for your reply.

**Reply by MichaReiser** [reply]
I'm glad you were able to figure it out, and thanks for sharing the solution. I didn't know about this option
